packages/paste-core/components/combobox/src/Combobox.tsx

```diff
@@ -138,6 +138,8 @@ const Combobox = React.forwardRef<HTMLInputElement, ComboboxProps>(
       iconColor = 'colorTextInverseWeak';
     }
 
+    const {ref: toggleButtonRef, ...toggleButtonProps} = getToggleButtonProps({tabIndex: 0, ref});
```

Comment:
Unlike `getInputProps`, `getToggleButtonProps` is _not_ an impure function. This means that we do not need to call it and spread it inline with the component. We can pass the forwarded ref to `getToggleButtonProps`, and pass the returned ref to `getInputProps`.